src/runtime/mod.rs

```diff
@@ -74,29 +72,17 @@ impl AsyncRuntime {
         self.spawn(fut);
     }
 
-    /// Run a future in the foreground, blocking on it completing.
-    ///
-    /// This will panic if called from a sychronous context when tokio is being used.
-    #[cfg(any(feature = "sync", test))]
+    #[cfg(test)]
     pub(crate) fn block_on<F, T>(self, fut: F) -> T
     where
         F: Future<Output = T>,
     {
-        #[cfg(all(feature = "tokio-runtime", not(feature = "async-std-runtime")))]
-        {
-            match TokioCallingContext::current() {
-                TokioCallingContext::Async(_handle) => {
-                    tokio::task::block_in_place(|| futures::executor::block_on(fut))
-                }
-                TokioCallingContext::Sync => {
-                    panic!("block_on called from tokio outside of async context")
-                }
-            }
-        }
+        match self {
```

Comment:
I don't think this (or `spawn`, for that matter) need the `match self` - and, thinking about it for a little more, I don't think `Runtime` needs to be an enum at all.  Most of the methods ignore the value in favor of `cfg` blocks, and it'll only ever have a single actual discriminant.

Reply:
yeah that's a good point, it's also a little confusing as the syntax of calling these methods (i.e. `RUNTIME.method()`) makes it appear that they're being called on the async runtimes directly. I refactored a bit to remove the enum and make the methods static functions within the `runtime` module.

Reply:
Very nice!